### PR TITLE
Consolidate color conversions, remove code duplication in texture scaler

### PR DESCRIPTION
**Pull request by hrydgard**
Attempt number 2.

This one is a bit less ambitious than the last one and I was a bit more careful. So hopefully no breakage this time...

### CODE REVIEW

File: Common/ColorConv.h [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+	return ((px >> 20) & 0x000F) | ((px >> 8) & 0x00F0) | ((px << 4) & 0x0F00) | ((px >> 16) & 0xF000);
+}
+
+inline u16 BGRA8888toRGBA5551(u32 px) {
+	return ((px >> 19) & 0x001F) | ((px >> 6) & 0x03E0) | ((px << 7) & 0x7C00) | ((px >> 16) & 0x8000);
+}
+
+inline u16 RGBA8888toRGBA5551(u32 px) {
+	return ((px >> 3) & 0x001F) | ((px >> 6) & 0x03E0) | ((px >> 9) & 0x7C00) | ((px >> 16) & 0x8000);
+}
```
Comment (unknownbrackets) [inline Common/ColorConv.h]: Do we still need/want these in the header here?  I think RGBA8888toRGBA5551 is no longer used externally, for example.  Probably better if we can centralize everything on using conversion funcs that we can move to SIMD.

And like RGBA2BGRA is also no longer used and already now using SSE (yay.)

-[Unknown]

File: Common/ColorConv.h [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+// convert image to 8888, parallelizable
+// TODO: Implement these in terms of the conversion functions below.
+void convert4444_gl(u16* data, u32* out, int width, int l, int u);
+void convert565_gl(u16* data, u32* out, int width, int l, int u);
+void convert5551_gl(u16* data, u32* out, int width, int l, int u);
+void convert4444_dx9(u16* data, u32* out, int width, int l, int u);
+void convert565_dx9(u16* data, u32* out, int width, int l, int u);
+void convert5551_dx9(u16* data, u32* out, int width, int l, int u);
```
Comment (unknownbrackets) [inline Common/ColorConv.h]: I wonder if we should keep these separate, or if it makes sense to have all the funcs use this format?  Hmm.

-[Unknown]

Comment (hrydgard) [inline Common/ColorConv.h]: Yeah, I couldn't decide..

File: Common/ColorConv.h (hunk 1 above)
Comment (hrydgard) [inline Common/ColorConv.h]: I'll see how many I can get rid of.

**Pull request merged**